src/api-service/__app__/onefuzzlib/pools.py

```diff
@@ -154,16 +160,87 @@ def stop_task(cls, task_id: UUID) -> None:
         for node in nodes:
             if node.state not in NodeState.ready_for_reset():
                 logging.info(
-                    "stopping task %s on machine_id:%s",
-                    task_id,
+                    "stopping machine_id:%s running task:%s",
                     node.machine_id,
+                    task_id,
                 )
-                node.state = NodeState.done
-                node.save()
+                node.stop()
+
+    def mark_tasks_stopped_early(self) -> None:
+        from .tasks.main import Task
+
+        for entry in NodeTasks.get_by_machine_id(self.machine_id):
+            task = Task.get_by_task_id(entry.task_id)
+            if isinstance(task, Task):
+                task.mark_failed(
+                    Error(
+                        code=ErrorCode.TASK_FAILED,
+                        errors=["node reimaged during task execution"],
+                    )
+                )
+            entry.delete()
+
+    def can_process_new_work(self) -> bool:
+        if not self.is_outdated():
+            logging.info(
+                "can_schedule old version machine_id:%s version:%s",
+                self.machine_id,
+                self.version,
+            )
+            self.stop()
+            return False
+
+        if self.delete_requested or self.reimage_requested:
+            logging.info(
+                "can_schedule should be recycled.  machine_id:%s", self.machine_id
+            )
+            self.stop()
+            return False
+
+        if self.scaleset_id is not None:
+
+            if ScalesetShrinkQueue(self.scaleset_id).should_shrink():
+                self.halt()
+                logging.info(
+                    "node scheduled to shrink.  machine_id:%s", self.machine_id
+                )
+                return False
+
+        return True
 
     def is_outdated(self) -> bool:
         return self.version != __version__
 
+    def send_message(self, message: NodeCommand) -> None:
+        stop_message = NodeMessage(
+            agent_id=self.machine_id,
+            message=message,
+        )
+        stop_message.save()
+
+    def to_reimage(self, done: bool = False) -> None:
+        if done:
+            if self.state not in NodeState.ready_for_reset():
+                self.state = NodeState.done
+
+        if not self.reimage_requested and not self.delete_requested:
+            self.reimage_requested = True
```

Comment:
Is this reset after the reimage ?

Reply:
It is now that #86 has been merged, I was able to enable it.